Formula/teip.rb

```diff
@@ -0,0 +1,28 @@
+class Teip < Formula
+  desc 'Masking tape to help commands "do one thing well"'
+  homepage "https://github.com/greymd/teip"
+  url "https://github.com/greymd/teip/archive/v2.3.0.tar.gz"
+  sha256 "4c39466613f39d27fa22ae2a6309ac732ea94fdbc8711ecd4149fc1ecdfbaedf"
+  license "MIT"
+  head "https://github.com/greymd/teip.git", branch: "main"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "oniguruma"
+  uses_from_macos "llvm" # for libclang
```

Comment:
```suggestion
  uses_from_macos "llvm" => :build # for libclang
```

Since `libclang` is used only by `bindgen`, a [build-time dependency](https://github.com/rust-onig/rust-onig/blob/main/onig_sys/Cargo.toml#L28) of `onig-sys`, this shouldn't be needed during runtime.